### DISCUSSION TITLE
HyperHDR hardware clarification

### DISCUSSION THREAD

**Discussion by jsv93** [awawa-dev/HyperSerialPico#16] [Q&A] [answered]
Looking into setting this up as using the HyperSerialPico config is much simpler. However, I have a question regading the HyperHDR hardware. Does a Rpi need to be used to host HyperHDR? Currently I have a small optiplex 3040 running linux with a few other services running inlcuding HyperHDR which has been fine. Just wondering if its neccesary to use a Rpi or if my current config will work.

Also with the other services running, mainly a Bitfocus satelite node, impact HyperSerialPico as it also uses a usb port?

**Comment by awawa-dev** [comment, marked as answer]
You don't need to have RPi, HyperSerialPico works on Mac (Intel, M1 I haven't tested), webOS, Windows PC, Linux ARM/x64 etc. because it uses a standard serial port protocol well supported by almost all systems (CDC) and does not even require special permissions in the system. As for resources, I haven't noticed any conflicts with other devices, but in the past there were problems with esp8266 that used ch340g (and not CDC) when using USB2.0 resources on RPi3 https://github.com/awawa-dev/HyperSPI/tree/5a17e197ecfaf3aecc2210b7912d1ca6db63fb8b : fighting for resources limited the maximum bandwidth of the USB2.0 grabber but we're talking here about the maximum bitrate when transmitting video reaching the limits specified by USB2.0 protocol. But this was almost certainly due to the limited hardware performance of the USB bus on the RPi3 combined with the poor performance and quality of ch340g drivers. Which led to the creation of HyperSPI. BTW HyperSPI support for Pico is almost ready but you would need RPi for HyperSPI.

**Reply by jsv93** [reply]
thats great then. I'll stay with the hardware config I have for the moment. The Optiplex 3040 has native usb3.0 so bus saturation shouldnt be an issue anyway.

Regarding HyperSPI vs HyperSerialPico is HyperSPI still the faster protocol? I assume so as its not using the USB bus at all. I have a few Pis spare so no doubt when HyperSPIPico comes out Ill cahnge right away.

**Reply by awawa-dev** [reply]
Don't let the Pico/rp2040 board plug confuse you: underneath it's still USB1.1, which in practice offers a maximum speed of around 4-5Mbps. So it is much faster than what we can achieve, e.g. with Arduino, but still slower than HyperSPI which works above 20MBps, and in addition SPI is "raw" protocol without any additional overhead: USB communication is more complicated.
